### PR TITLE
Fix/newsletter reply to setting validation

### PR DESCRIPTION
**Pull request by enejb**
## Proposed changes:
This improves the way that we validate reply_to by using the new method. 
The new method was not yet shipped and the comment type is only supported by .com simple sites for now. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:
Go to /wp-admin/admin.php?page=jetpack#/newsletter
And save the Reply to setting. 
Notice it saves as expected.

### CODE REVIEW

**Comment by jeherve** [comment]
Since you're asking for a review from Zap, maybe you meant to use the "Needs Team Review" label instead of "Needs Review"? "Needs Review" is meant to request an additional review from Crew, when you'd like an additional gut check that your team cannot provide.

If you'd like this PR to be reviewed by Crew (or probably by your team :) ), could you add some testing instructions to make that easier?

Thanks!

**Comment by simison** [comment]
@jeherve feels like these labels would be clearer:

- Needs review
- Needs Crew Review

**Comment by jeherve** [comment]
@simison Yes, this has come up a few times in the past couple of years. I would agree. It makes more sense now that we no longer require approvals from the Approvers group. Last we talked about it was here:
p9dueE-8aF-p2#comment-10348

File: projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php
Comment (pkuliga) [inline projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php:3413]: This new validation method also allows the "comment" value. Are we ok with it?

https://github.com/Automattic/jetpack/blob/ce1a7890b6abf33bedf96f3ce96fe43a4bf7d428/projects/plugins/jetpack/modules/subscriptions/class-settings.php#L25-L31

Comment (simison) [reply projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php:3413]: I think it's fine since we're looking into making the comments work too (https://github.com/Automattic/jetpack/issues/37369); there's no UI in Jetpack to switch to that value anyway yet, but this will make testing easier.

**Review by pkuliga — APPROVED** [review]

**Pull request merged**
